src/gluonts/dataset/loader.py

```diff
@@ -28,55 +30,37 @@
 DataBatch = Dict[str, Any]
 
 
-class BatchBuffer:
-    def __init__(
-        self, batch_size: int, ctx: mx.Context, dtype: DType = np.float32
-    ) -> None:
-        self._buffers: Dict[Any, List[Any]] = defaultdict(list)
+def batcher(iterable, batch_size):
+    while True:
+        items = list(itertools.islice(iterable, batch_size))
+        if not items:
+            break
+        yield items
+
+
+class BatchStacker:
```

Comment:
These classes are nice, but could use a docstring to briefly summarize what they do: “An Iterable type, that given a sequence of X yields a sequence of Y”

Reply:
I'm also not too sure about the names. Let me improve these a bit.